### DISCUSSION TITLE
Vector Valued Variables in MultApp

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#20858] [Q&A Modules: General] [answered]
I am looking to use multi-app to pull velocity (vector) data from a transient Naiver Stokes solution into a separate app that imposes the values of the transient solution on to a steady state domain. For instance if I'm using the transient values for the initial conditions for a steady state solutions. Can this be accomplished by use of an existing kernel, aux kernel, or anything? In my case the two mesh domains share the same dimensions and mesh densities.

**Comment by GiudGiud** [comment, marked as answer]
Hello

This is Navier Stokes finite element or finite volume ?
If they use the same mesh you can use the MultiAppCopyTrasnfer on the **components** of the velocity.
https://mooseframework.inl.gov/source/transfers/MultiAppCopyTransfer.html
In FE, you can use this https://mooseframework.inl.gov/source/auxkernels/VectorVariableComponentAux.html to retrieve velocity components. 

Then you can initialize the velocity component by component.
In FV it's simple, transfer directly to the right variable velocity component.
In FE you ll have to create a new object, a BuildVectorVariableAux, similar to this
https://mooseframework.inl.gov/source/auxkernels/BuildArrayVariableAux.html

Let us know if you need help with this
An example of a VectorAuxkernel is this one:
https://github.com/idaholab/moose/blob/next/framework/src/auxkernels/VectorFunctionAux.C

Guillaume

**Reply by TLWise** [reply]
Thank you for the reply. I am using FE. I will review the referenced links you provided, and reach out to you for additional guidance if I run into further roadblocks to my understanding.

**Reply by TLWise** [reply]
I executed  MultiAppCopyTransfer on of the sample problems that uses MultiAppCopyTransfer, I also attempted to use it in a code that I developed. I keep getting the same error "Unable to locate object" each time. Is it possible that I will need to do and update and remake prior to being able to run this particular command?

**Reply by GiudGiud** [reply]
What are you trying to copy? The vector variable or the variable components?

**Reply by TLWise** [reply]
I was able to get MultiAppCopyTransfer to work properly when I used the components of a the vector or a scalar variable like Temperature. I must have been attempting to use velocity initially. Thanks
